core/wasm/src/SegmentPool.kt

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.io
+
+internal actual object SegmentPool {
+    actual val MAX_SIZE: Int = 0
+
+    actual val byteCount: Int = 0
+
+    actual fun take(): Segment = Segment()
```

Comment:
No real pool in WASM?

Reply:
The pooling is only supported for JVM target at the moment. I'm thinking of adding support for other targets later.